README.md

```diff
@@ -67,8 +67,9 @@ const [{}, makeRequest] = useRequest();
 | timeout | Request allowed duration in `milliseconds`. When request duration exceeds this value, the request will be aborted. | `number` |   |
 | forceRefetch | It determines if stored/cached value by request, should be returned or new data should be fetched from server  | `boolean` |  `false` |
 | isRelative | It specify that the `url` passed to `makeRequest` is a relative path and force the base URL passed to either `useRequest` hook or `RequestProvider` to be prepend to the `makeRequest` function `url` regardless of if it is an absolute url or a relative path. There would most likely not be a need to set this property, as the library can determine whether to use the baseUrl based on what is passed to the `makeRequest` function. | `boolean` | `false` |
-| errorMessage | Request error message. The library tries to get the error message from the response payload and returns it in the state message prop, but this override any error message gotten from the response payload | `string` |     |
-| successMessage | Response success message. The library tries to get the success message from the response payload and returns it in the state message prop, but this override any success message gotten from the response payload | `string` |   |
+| errorMessage | Request error message. The library tries to get the error message from the response payload and returns it in the state message prop, but this override any error message gotten from the response payload. It is `deprecated`, use `metadata` to set the `errorMessage` instead. | `string` |     |
+| successMessage | Response success message. The library tries to get the success message from the response payload and returns it in the state message prop, but this override any success message gotten from the response payload. It is `deprecated`, use `metadata` to set the `successMessage` instead. | `string` |   |
+| metadata | Holds any data that needs to be made available in any of the listed request callbacks. callbacks: `onSuccess`, `onError` & `interceptors` | `object` |   |
```

Comment:
metadata | This is an object  where you can add any request metadata or information that is needed to be used elsewhere, so the content of this metadata will be made available in any of the listed request callbacks. Callbacks: `onSuccess`, `onError` & `interceptors` | `object` |   |